library_generation/owlbot/src/apply_repo_templates.py

```diff
@@ -0,0 +1,59 @@
+"""
+This script parses an owlbot.py file, specifically the call to `java.common_templates` in
+order to extract the excluded files so it can be called with a custom template path
+pointing to the templates hosted in `sdk-platform-java/library_generation/owlbot/templates`.
+Briefly, this wraps the call to synthtool's common templates using a custom template folder.
+"""
+
+import os
+from collections.abc import Sequence
+from absl import app
+from synthtool.languages.java import common_templates
+from pathlib import Path
+import re
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+repo_templates_path = os.path.join(script_dir, '..', 'templates', 'java_library')
+
+def apply_repo_templates(owlbot_py_path: str, monorepo: bool) -> None:
+  excludes = []
+  with open(owlbot_py_path) as contents:
+    excludes += parse_template_excludes(contents.read())
+  print(f'repo_templates_path: {repo_templates_path}')
+  print(f'excludes: {excludes}')
+  common_templates(
+      excludes=excludes,
+      template_path=Path(repo_templates_path),
+      monorepo=monorepo
+  )
+
+
+def parse_template_excludes(owlbot_py_contents: str) -> str:
```

Comment:
`java.common_templates` lives in [`synthtool`](https://github.com/googleapis/synthtool/blob/5e1fb2032fa44bc170677b38713023b4fec51a4e/synthtool/languages/java.py#L476)

If we want to change the implementation of that function we would need to transfer synthtool to sdk-platform-java (which is an actual option).

Also from the PR description, we could modify all owlbot.py files (and its template) to call `our_own_owlbot.common_templates` instead of `java.common_templates`

cc: @suztomo

Reply:
As discussed with @suztomo, I'll use the configuration yaml to have a list of exclusions

Reply:
I modified the implementation to use the configuration yaml